test/e2e/framework/service_util.go

```diff
@@ -525,6 +528,44 @@ func (j *ServiceTestJig) newRCTemplate(namespace string) *v1.ReplicationControll
 	return rc
 }
 
+func (j *ServiceTestJig) AddRCAntiAffinity(rc *v1.ReplicationController) {
```

Comment:
It needs access to the jig to get the correct labels.  I could pass the jig as an argument, or the test could grab the labels from the jig and pass them as an argument.  In the end I left it here because the tweak requires requires a bunch of internal knowledge about all the rest of the stuff the jig does.